### PR TITLE
[BPK-1422] Add props to mobile-scroll-indicator to allow styling

### PR DESCRIPTION
**Pull request by shaundon**
## Details
* Two new props in `bpk-mobile-scroll-container` and `bpk-horizontal-nav` for adding class names to the leading and trailing scroll indicators.
* This allows users to use `BpkHorizontalNav` on non-white backgrounds and still have the indicators look 👌 👌 👌 .
* Abstracted the logic for creating the scroll indicators away from `bpk-mobile-scroll-indicator` into a mixin that users can make use of when applying these props.
* There's an example of this in the storybook.

### Before
screen shot 2018-05-02 at 11 44 17

### After
screen shot 2018-05-02 at 11 42 34

### CODE REVIEW

File: packages/bpk-component-horizontal-nav/readme.md [hunk 1]
```diff
@@ -62,10 +62,12 @@ export default class App extends Component {
...
 
-| Property    | PropType | Required | Default Value |
-| ----------- | -------- | -------- | ------------- |
-| children    | node     | true     | -             |
-| className   | string   | false    | null          |
+| Property                          | PropType | Required | Default Value |
+| --------------------------------- | -------- | -------- | ------------- |
+| children                          | node     | true     | -             |
+| className                         | string   | false    | null          |
+| leadingScrollIndicatorClassName   | string   | false    | null          |
```
[the remark this answers is not in the document]
Comment (matthewdavidson) [reply packages/bpk-component-horizontal-nav/readme.md]: Its a `MobileScrollIndicator` component, no point adding redundant words when there is context that can be inferred.

It's the same as:
```
<Button
  buttonId
  buttonOnChange
/>
```
whereas this is better:
```
<Button
  id 
  onChange
/>
```

[the remark this answers is not in the document]
Comment (matthewdavidson) [reply packages/bpk-component-horizontal-nav/readme.md]: **Update:** I realise now that I commented on the horizontal nav. Perhaps including scroll in the name provides more context. Im actually happy with the names in that case.

Maybe stick with `leadingScrollIndicatorClassName` for nav and `leadingIndicatorClassName` for mobile scroll container?

[the remark this answers is not in the document]
Comment (shaundon) [reply packages/bpk-component-horizontal-nav/readme.md]: Ah I see. Yeah your reasoning makes sense - I'll change it for the mobile scroll container 😄

File: packages/bpk-component-mobile-scroll-container/src/__snapshots__/BpkMobileScrollContainer-test.js.snap [hunk 2]
```diff
@@ -44,6 +44,28 @@ exports[`BpkMobileScrollContainer should render correctly with a custom "classNa
 </div>
 `;
 
+exports[`BpkMobileScrollContainer should render correctly with a custom "leadingScrollIndicatorClassName" attribute 1`] = `
```
[the remark this answers is not in the document]
Comment (shaundon) [reply packages/bpk-component-mobile-scroll-container/src/__snapshots__/BpkMobileScrollContainer-test.js.snap]: Ah, I just figured out why. These classes are only applied when the indicator should display, like when content goes off screen and it's scrollable. Not quite sure of the best way around that, perhaps something with Enzyme to mock some scrolling?

File: packages/bpk-component-horizontal-nav/readme.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (shaundon) [reply packages/bpk-component-horizontal-nav/readme.md]: Done!

**Comment by matthewdavidson** [comment]
Just noticed that we'll have to update the bar chart to support those new props too 👍

File: packages/bpk-component-mobile-scroll-container/src/__snapshots__/BpkMobileScrollContainer-test.js.snap (hunk 2 above)
[the remark this answers is not in the document]
Comment (matthewdavidson) [reply packages/bpk-component-mobile-scroll-container/src/__snapshots__/BpkMobileScrollContainer-test.js.snap]: The issue here is that this test is providing no value and is confusing - i.e. it's the same as having two tests with different names but the same assertions?

We either do as you say and try to trick the scroll container to think it's children are overflowing in the test and assert accordingly or we delete the tests.

**Pull request merged**